app/src/main/kotlin/com/x8bit/bitwarden/ui/vault/feature/vault/util/VaultDataExtensions.kt

```diff
@@ -44,15 +45,36 @@ fun VaultData.toViewState(
     vaultFilterType: VaultFilterType,
     restrictItemTypesPolicyOrgIds: List<String>?,
 ): VaultState.ViewState {
-
-    val filteredCipherViewListWithDeletedItems =
+    val filteredAllCipherViewListWithDeletedItems =
         decryptCipherListResult
             .successes
+            .plus(
+                elements = decryptCipherListResult
+                    .failures
+                    .map { cipher ->
+                        cipher.toFailureCipherListView()
+                    },
+            )
             .applyRestrictItemTypesPolicy(restrictItemTypesPolicyOrgIds ?: emptyList())
             .toFilteredList(vaultFilterType)
 
-    val filteredCipherViewList = filteredCipherViewListWithDeletedItems
+    val filteredCipherViewList = filteredAllCipherViewListWithDeletedItems
+        .filter { it.deletedDate == null }
+
+    val filteredSuccessCipherViewList = decryptCipherListResult
+        .successes
+        .filter { it.deletedDate == null }
+        .applyRestrictItemTypesPolicy(restrictItemTypesPolicyOrgIds ?: emptyList())
+        .toFilteredList(vaultFilterType)
+
+    val filteredFailureCipherViewList = decryptCipherListResult
+        .failures
         .filter { it.deletedDate == null }
+        .map { cipher ->
+            cipher.toFailureCipherListView()
+        }
+        .applyRestrictItemTypesPolicy(restrictItemTypesPolicyOrgIds ?: emptyList())
+        .toFilteredList(vaultFilterType)
```

Comment:
Similar to my previous comment, it would be nice to extract this common filtering logic into an ext function.